ui/src/components/Pin.tsx

```diff
@@ -5,13 +5,19 @@ import styled from "styled-components";
 
 import resting from "../assets/resting.svg";
 import unfocused from "../assets/unfocused.svg";
+import { device } from "../styles/device";
 import { Story } from "../types";
 import { PinPreview } from "./PinPreview";
 
 const StyledPopup = styled(Popup)`
   width: 392px;
   height: 467px;
   box-shadow: 0px 0px 25px 5px rgba(0, 0, 0, 0.2);
+  @media ${device.mobile} {
+    width: fit-content !important;
```

Comment:
Did you test without `!important` for any of these declarations? Do they all require it?